### ISSUE TITLE
[CRASH] Core dump accessing memory after being freed

### ISSUE THREAD

**Issue by danpascu** [OpenSIPS/opensips#1697]
**OpenSIPS version you are running**

```
version: opensips 3.0.0-beta (x86_64/linux)
flags: STATS: On, DISABLE_NAGLE, USE_MCAST, SHM_MMAP, Q_MALLOC, F_MALLOC, HP_MALLOC, DBG_MALLOC, CC_O0, FAST_LOCK-ADAPTIVE_WAIT
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144, MAX_LISTEN 16, MAX_URI_SIZE 1024, BUF_SIZE 65535
poll method support: poll, epoll, sigio_rt, select.
git revision: 1e14844dd
main.c compiled on  with gcc 8
```

**Crash Core Dump**

[logs.txt](https://github.com/OpenSIPS/opensips/files/3177796/logs.txt)

**Describe the traffic that generated the bug**

Received a SUBSCRIBE that is handled by rls

**To Reproduce**

Cannot reproduce. The crash was random. I believe the crash is a side effect of some other memory corruption (like for example overwriting the cmd argument type to be of type pve after it was determined that it is not pve and the pve was released)

**OS/environment information**
 - Operating System:  Debian sid

**Comment by danpascu** [comment]
Happened again. Is still random, I cannot reproduce it at will with a particular message. Seems to have started happening after I started playing with the list_statistics MI command. Before this it ran for weeks without a crash with the same SUBSCRIBE traffic as it has now. This is a test environment with just a few controlled clients running on it.

**Comment by danpascu** [comment]
And it happened again this time with a PUBLISH. I think my assumption that it is somehow related to using list_statistics is false. I also updated to HEAD from git today after I fixed the list_statistics command. This is when it started happening. Before this I was running 08cff3486e8ad9e7f4fc52cdb202c3201a3435ae (Revert "tracer: always load the trace api") so it might be related to some commit since then.

**Comment by danpascu** [comment]
It seems I was wrong. This can be reproduced 100%. All I need to do is send a PUBLISH. I used git bisect to find the patch that introduced the problem:

```
307e8c8ad6ce9bd263a122c9e0a8164cf59a131e is the first bad commit
commit 307e8c8ad6ce9bd263a122c9e0a8164cf59a131e
Author: Vlad Patrascu <vladp@opensips.org>
Date:   Tue May 7 19:05:24 2019 +0300

    Fix the CMD_PARAM_NO_EXPAND parameter flag
    
    The fixup of parameters flagged wih CMD_PARAM_NO_EXPAND would fail if
    the string contained a '$' character not followed by a valid variable.

:100644 100644 fa3da607f0cad7215ca2d2b91756208a29ea4946 63618fb5211b1b90345323d4ef78a98e2bb70b68 M      mod_fix.c
```

**Comment by danpascu** [comment]
Any comment on this? Because of that patch now opensips crashes on any presence related traffic. The patch should either be amended of it should be reversed until a better solution is found.

**Comment by bogdan-iancu** [comment]
Let me check it

**Comment by bogdan-iancu** [comment]
@danpascu , you say you can easily get it reproduced by pushing a PUBLISH into `handle_publish()` function ?

**Comment by danpascu** [comment]
Yes . My code is this (after authenticating the request):

```
        if (t_newtran()) {
            handle_publish();
        } else {
            sl_reply_error();
        }
        exit;
```

But I do not think it matters much how I reproduce it. Just check the ASAN trace I attached. It gives you all the information: where the memory was allocated, where it was released and where it was accessed after it was released. The problem is accessing memory after it is released, it just happens that it is triggered by a PUBLISH, but from my experience it is also triggered by SUBSCRIBE and who knows what else. My SIP client changes the presence state and it publishes it right before it makes a call, so that is the first thing that hits the proxy when I attempt to make a call.
